packages/components/src/hooks/use-constructor.js

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+export const useConstructor = (callBack = () => {}) => {
+    const [hasBeenCalled, setHasBeenCalled] = React.useState(false);
```

Comment:
- Yes, `useRef` is a better choice here. Thanks for suggesting 🙏 
- We can do the same with `useEffect`. But we can avoid a render cycle by using `useConstructor` instead of `useEffect`.

Reply:
Ah I think you're right. 👍